README.md
# motorcortex-backgrounds

## Demo
[Check it out here](https://kissmybutton.github.io/motorcortex-backgrounds/demo/index.html)

## Installation

```bash
$ npm install --save @kissmybutton/motorcortex-backgrounds
# OR
$ yarn add @kissmybutton/motorcortex-backgrounds
```

## Loading

```javascript
import {loadPlugin} from "@kissmybutton/motorcortex";
import BackgroundsDefinition from "@kissmybutton/motorcortex-backgrounds";
const Plugin = loadPlugin(BackgroundsDefinition);
```

# Create incident

## BgOpener

```javascript
const BgOpener = new Plugin.BgOpener(
  {
    width: 512,
    height: 288,
    bgUrl:"https://kissmybutton.github.io/motorcortex-slides/demo/kissmybutonbg.jpg",
    timing:0.3,
    easing:"easeInQuad",
    exitStart:2000
  },
  {
    selector: ".container1"
  }
);
```

### BgOpener Attrs

| Name        | Are           | Values  |
| ------------- |:-------------:| -----:|
| width     | width of incident  | num |
| height     | height of incident  | num |
| bgUrl     | url of the bg image  | string |
| easing     | easing of the animation  | string |
| exitStart     | the millisecond that we want to start the bg exiting  | num |


## TwoSidesReveal

```javascript
const TwoSidesReveal = new Plugin.TwoSidesReveal(
  {
    width: 512,
    height: 288,
    bgUrl:
    "https://kissmybutton.github.io/motorcortex-slides/demo/kissmybutonbg.jpg",
    easing:"easeInQuad",
  },
  {
    selector: ".container2"
  }
);
```

### TwoSidesReveal Attrs

| Name        | Are           | Values  |
| ------------- |:-------------:| -----:|
| width     | width of incident  | num |
| height     | height of incident  | num |
| bgUrl     | url of the bg image  | string |
| easing     | easing of the animation  | string |



## RowReveal

```javascript
const RowReveal = new Plugin.RowReveal(
  {
    width: 512,
    height: 288,
    bgUrl:
    "https://kissmybutton.github.io/motorcortex-slides/demo/kissmybutonbg.jpg",
    easing:"easeOutQuart",
    exitStart:2000,
    bgOut: true
  },
  {
    selector: ".container3"
  }
);
```

### RowReveal Attrs

| Name        | Are           | Values  |
| ------------- |:-------------:| -----:|
| width     | width of incident  | num |
| height     | height of incident  | num |
| bgUrl     | url of the bg image  | string |
| easing     | easing of the animation  | string |
| exitStart     | the millisecond that we want to start the bg exiting  | num |
| bgOut     | set true if we like to have a bg exit  | boolean |


## ColumnReveal

```javascript
const ColumnReveal = new Plugin.ColumnReveal(
  {
    width: 512,
    height: 288,
    bgUrl:
    "https://kissmybutton.github.io/motorcortex-slides/demo/kissmybutonbg.jpg",
    easing:"easeOutQuart",
    exitStart:2000,
    bgOut: true
  },
  {
    selector: ".container4"
  }
);
```

### ColumnReveal Attrs

| Name        | Are           | Values  |
| ------------- |:-------------:| -----:|
| width     | width of incident  | num |
| height     | height of incident  | num |
| bgUrl     | url of the bg image  | string |
| easing     | easing of the animation  | string |
| exitStart     | the millisecond that we want to start the bg exiting  | num |
| bgOut     | set true if we like to have a bg exit  | boolean |



## Grid

```javascript
const Grid = new Plugin.Grid(
  {
    width: 512,
    height: 288,
    color: "#000",
    columns: 6,
    rows : 4,
  },
  {
    selector: ".container6"
  }
);
```

### Grid Attrs

| Name        | Are           | Values  |
| ------------- |:-------------:| -----:|
| width     | width of incident  | num |
| height     | height of incident  | num |
| color     | the color of the background  | hex,rgb |
| columns     | how many columns the grid will have  | num |
| rows     | how many rows the grid will have  | num |



## ThreeSidesReveal

```javascript
const ThreeSidesReveal = new Plugin.ThreeSidesReveal(
  {
    width: 512,
    height: 288,
    bgUrl: "https://kissmybutton.github.io/motorcortex-slides/demo/bg3.jpg",
    overlayColor: "#ff0000",
    grid: true,
    gridDuration:1,
    gridColor: "#000",
    columns: 6,
    rows : 4,
  },
  {
    selector: ".container5",
  }
);
```

### ThreeSidesReveal Attrs

| Name        | Are           | Values  |
| ------------- |:-------------:| -----:|
| width     | width of incident  | num |
| height     | height of incident  | num |
| bgUrl     | url of the bg image  | string |
| overlayColor   | overlay color of background  | hex,rgb |
| grid     | set true if we like to have a gid  | boolean |
| gridDuration     | grid incident duration  | num |
| gridColor     | the color of the background  | hex,rgb |
| columns     | how many columns the grid will have  | num |
| rows     | how many rows the grid will have  | num |


CHANGELOG.md
# Changelog

### [0.0.12](https://www.github.com/kissmybutton/motorcortex-backgrounds/compare/v0.0.11...v0.0.12) (2021-07-01)


### Bug Fixes

* remove attribute from incidents ([cc06926](https://www.github.com/kissmybutton/motorcortex-backgrounds/commit/cc06926ca585fc9a437e00584bb44e6074ccf871))

### 0.0.11 (2021-04-16)


### Bug Fixes

* fix package json ([9efb051](https://www.github.com/kissmybutton/motorcortex-backgrounds/commit/9efb051322bd579b615a59bea13007720f003765))
* release 0.0.11 ([cb5b42f](https://www.github.com/kissmybutton/motorcortex-backgrounds/commit/cb5b42faa8af1312bc4d6e89b5198097f8dae798))


demo/index.js
import { HTMLClip, loadPlugin } from "@kissmybutton/motorcortex";
import Player from "@kissmybutton/motorcortex-player";
import BackgroundsDefinition from "../dist/motorcortex-backgrounds.umd";
const Plugin = loadPlugin(BackgroundsDefinition);

const clip = new HTMLClip({
  css: `
    .container {
      overflow: hidden;
      width:512px;
      height:288px;
    }

    .wrapper{
      width:1728px;
      height:872px;
      display:flex;
      font-family: 'Poppins', sans-serif;
      display: grid;
      grid-template-columns: repeat(3, 1fr);
      justify-items: center;
      align-items: center;
    }

    .flex{
      display: flex;
      justify-content: center;
      background-color: #bfbfbf;
      height: 100%;
      width: 100%;
    }
  `,
  html: ` 
    <div class="flex">
      <div class="wrapper">
        <div class="container container1"></div>
        <div class="container container2"></div>
        <div class="container container3"></div>
        <div class="container container4"></div>
        <div class="container container5"></div>
        <div class="container container6"></div>
      </div>
    </div>
  `,
  host: document.getElementById("clip"),
  containerParams: {
    width: "1280px",
    height: "720px"
  },
  id: "root",
  fonts: [
    {
      type: `google-font`,
      src: `https://fonts.googleapis.com/css2?family=Poppins:ital,wght@0,100;0,400;0,500;0,800;1,700;1,800;1,900&display=swap`
    }
  ]
});

const BgOpener = new Plugin.BgOpener(
  {
    width: 512,
    height: 288,
    bgUrl:
      "https://kissmybutton.github.io/motorcortex-backgrounds/demo/assets/kissmybutonbg.jpg",
    easing: "easeInQuad",
    exitStart: 2000
  },
  {
    selector: ".container1"
  }
);

const TwoSidesReveal = new Plugin.TwoSidesReveal(
  {
    width: 512,
    height: 288,
    bgUrl:
      "https://kissmybutton.github.io/motorcortex-backgrounds/demo/assets/kissmybutonbg.jpg",
    easing: "easeInQuad"
  },
  {
    selector: ".container2"
  }
);

const RowReveal = new Plugin.RowReveal(
  {
    width: 512,
    height: 288,
    bgUrl:
      "https://kissmybutton.github.io/motorcortex-backgrounds/demo/assets/kissmybutonbg.jpg",
    timing: 1,
    easing: "easeOutQuart",
    exitStart: 2000,
    bgOut: true
  },
  {
    selector: ".container3"
  }
);

const ColumnReveal = new Plugin.ColumnReveal(
  {
    width: 512,
    height: 288,
    bgUrl:
      "https://kissmybutton.github.io/motorcortex-backgrounds/demo/assets/kissmybutonbg.jpg",
    timing: 1,
    easing: "easeOutQuart",
    exitStart: 2000,
    bgOut: true
  },
  {
    selector: ".container4"
  }
);

const ThreeSidesReveal = new Plugin.ThreeSidesReveal(
  {
    width: 512,
    height: 288,
    bgUrl:
      "https://kissmybutton.github.io/motorcortex-backgrounds/demo/assets/bg3.jpg",
    overlayColor: "#ff0000",
    grid: false,
    // gridDuration: 5000,
    gridColor: "#000",
    columns: 6,
    rows: 4
  },
  {
    selector: ".container5"
  }
);

const Grid = new Plugin.Grid(
  {
    width: 512,
    height: 288,
    color: "#000",
    columns: 6,
    rows: 4,
    timing: 1
  },
  {
    selector: ".container6"
  }
);

clip.addIncident(BgOpener, 0);
clip.addIncident(TwoSidesReveal, 0);
clip.addIncident(RowReveal, 0);
clip.addIncident(ColumnReveal, 0);
clip.addIncident(ThreeSidesReveal, 0);
clip.addIncident(Grid, 0);

new Player({ clip, timeFormat: "ms" });
